app/src/androidTest/java/com/github/geohunt/app/ui/components/profile/ProfilePageTest.kt

```diff
@@ -1,127 +1,236 @@
 package com.github.geohunt.app.ui.components.profile
 
+import android.app.Application
 import androidx.compose.ui.test.assertIsDisplayed
 import androidx.compose.ui.test.assertIsNotDisplayed
 import androidx.compose.ui.test.junit4.createComposeRule
 import androidx.compose.ui.test.onNodeWithTag
 import androidx.compose.ui.test.onNodeWithText
 import androidx.compose.ui.test.performClick
-import com.github.geohunt.app.mocks.InstantLazyRef
-import com.github.geohunt.app.mocks.MockChallengeClass
-import com.github.geohunt.app.mocks.MockLazyRef
-import com.github.geohunt.app.mocks.MockUser
-import com.github.geohunt.app.model.LazyRef
-import com.github.geohunt.app.model.database.api.Challenge
-import com.google.android.gms.tasks.Tasks
+import com.github.geohunt.app.mocks.MockAuthRepository
+import com.github.geohunt.app.data.repository.AppContainer
+import com.github.geohunt.app.data.repository.AuthRepositoryInterface
+import com.github.geohunt.app.data.repository.ChallengeRepositoryInterface
+import com.github.geohunt.app.data.repository.FollowRepositoryInterface
+import com.github.geohunt.app.data.repository.UserRepositoryInterface
+import com.github.geohunt.app.mocks.MockChallenge
+import com.github.geohunt.app.mocks.MockChallengeRepository
+import com.github.geohunt.app.mocks.MockFollowRepository
+import com.github.geohunt.app.mocks.MockUserRepository
+import com.github.geohunt.app.model.Challenge
+import com.github.geohunt.app.model.Claim
+import com.github.geohunt.app.model.User
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.withContext
+import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
+import java.time.LocalDateTime
+import java.util.concurrent.CompletableFuture
 
 class ProfilePageTest {
     @get:Rule
     val testRule = createComposeRule()
 
-    @Test
-    fun showsLoadingIfNotReady() {
-        testRule.setContent {
-            ProfilePage(user = InstantLazyRef("1", null))
-        }
-        testRule.onNodeWithTag("progress").assertIsDisplayed()
+    var appContainer: AppContainer? = null
```

Comment:
Doesn't `lateinit var` solve the problem of having `null` as the initial value ?

Reply:
Maybe, but I do not want to bother with that.

Reply:
It's test no-one check them. Plus I am pretty sure `lateinit var` only works within the context of a field within a class